internal/factsengine/gatherers/sapservices.go

```diff
@@ -0,0 +1,180 @@
+package gatherers
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+	"github.com/trento-project/agent/pkg/factsengine/entities"
+)
+
+type SapServicesStartupKind string
+
+const (
+	SapServicesSystemdStartup  SapServicesStartupKind = "systemctl"
+	SapServicesSapstartStartup SapServicesStartupKind = "sapstartsrv"
+	sapServicesDefaultPath                            = "/usr/sap/sapservices"
+	SapServicesGathererName                           = "sapservices"
+)
+
+// nolint:gochecknoglobals
+var (
+	SapServicesParsingError = entities.FactGatheringError{
+		Type:    "sap-services-parsing-error",
+		Message: "error parsing the sap services file",
+	}
+	SapServicesFileError = entities.FactGatheringError{
+		Type:    "sap-services-parsing-error",
+		Message: "error reading the sap services file",
+	}
+	SapstartSIDExtractionPattern = regexp.MustCompile(`(?s)pf=([^[:space:]]+)/(.*?)_`)
+	SystemdSIDExtractionPattern  = regexp.MustCompile(`(?s)start SAP(.*?)_.* `)
+)
+
+type SapServicesEntry struct {
+	SID     string                 `json:"sid"`
+	Kind    SapServicesStartupKind `json:"kind"`
+	Content string                 `json:"content"`
+}
+
+func systemdStartup(sapServicesContent string) bool {
+	return strings.Contains(sapServicesContent, "systemctl")
+}
+
+func sapstartStartup(sapServicesContent string) bool {
+	return strings.Contains(sapServicesContent, "sapstartsrv")
+}
+
+func extractSIDFromSystemdService(sapServicesContent string) string {
+	matches := SystemdSIDExtractionPattern.FindStringSubmatch(sapServicesContent)
+	if len(matches) != 2 {
+		return ""
+	}
+	return matches[1]
+}
+
+func extractSIDFromSapstartService(sapServicesContent string) string {
+	matches := SapstartSIDExtractionPattern.FindStringSubmatch(sapServicesContent)
+	if len(matches) != 2 {
+		return ""
+	}
+	return matches[1]
+}
+
+type SapServices struct {
+	fs               afero.Fs
```

Comment:
Do you need afero if you pass a fixed file path?
For example, in the `groups` gatherer you implemented without it